integration-tests/cli/csv-parser-orders.it.js

```diff
@@ -14,161 +14,87 @@ describe('CSV and CLI Tests', () => {
-
-      exec(
-        `${binPath} -i ${csvFilePath} -o ${jsonFilePath} -t returninfo`,
-        (cliError, stdout, stderr) => {
-          expect(cliError && stderr).toBeFalsy()
 
-          fs.readFile(jsonFilePath, { encoding: 'utf8' }, (error, data) => {
-            expect(error).toBeFalsy()
-            expect(JSON.parse(data)).toEqual(expectedResult)
-            done()
-          })
-        }
+      await exec(
+        `${binPath} -i ${csvFilePath} -o ${jsonFilePath} -t returninfo`
       )
+      const data = await fs.readFile(jsonFilePath, { encoding: 'utf8' })
+      expect(JSON.parse(data)).toMatchSnapshot()
     })
   })
 
   describe('CLI logs specific errors', () => {
-    it('on faulty CSV format', done => {
+    it('on faulty CSV format', async () => {
       const csvFilePath = path.join(samplesFolder, 'faulty-sample.csv')
       const jsonFilePath = tmp.fileSync().name
-
-      exec(
-        `${binPath} -i ${csvFilePath} -o ${jsonFilePath} -t returninfo`,
-        (error, stdout, stderr) => {
-          expect(error.code).toBe(1)
-          expect(stdout).toBeFalsy()
-          expect(stderr.match(/Row length does not match headers/)).toBeTruthy()
-          done()
-        }
-      )
+      try {
+        await exec(
+          `${binPath} -i ${csvFilePath} -o ${jsonFilePath} -t returninfo`
+        )
+      } catch (error) {
+        expect(error.code).toBe(1)
+        expect(String(error)).toMatch(/Row length does not match headers/)
+      }
     })
 
-    it('on missing return-info headers', done => {
+    it('on missing return-info headers', async () => {
       const csvFilePath = path.join(
         samplesFolder,
         'return-info-error2-sample.csv'
       )
       const jsonFilePath = tmp.fileSync().name
 
-      exec(
-        `${binPath} -i ${csvFilePath} -o ${jsonFilePath} -t returninfo`,
-        (error, stdout, stderr) => {
-          expect(error.code).toBe(1)
-          expect(stdout).toBeFalsy()
-          expect(stderr).toMatch(/Required headers missing: 'orderNumber'/)
-          done()
-        }
-      )
+      try {
+        await exec(
+          `${binPath} -i ${csvFilePath} -o ${jsonFilePath} -t returninfo`
+        )
+      } catch (error) {
+        expect(error.code).toBe(1)
+        expect(String(error)).toMatch(/Required headers missing: 'orderNumber'/)
+      }
     })
 
-    it('on missing line-item-state headers', done => {
+    it('on missing line-item-state headers', async () => {
       const csvFilePath = path.join(samplesFolder, 'return-info-sample.csv')
       const jsonFilePath = tmp.fileSync().name
 
-      exec(
-        `${binPath} -i ${csvFilePath} -o ${jsonFilePath} -t lineitemstate`,
-        (error, stdout, stderr) => {
-          expect(error.code).toBe(1)
-          expect(stdout).toBeFalsy()
-          expect(stderr).toMatch(
-            /Required headers missing: 'fromState,toState'/
-          )
-          done()
-        }
-      )
+      try {
+        await exec(
```

Comment:
Nice! Just for completeness: this is one way of solving the problem. You could also just `await expect(...).rejects.toEqual(...)`.

Reply:
Good point, I actually looked into that way but didnt get the string matching to work with `toEqual()` since its rejects to an error Object. However, it worked fine with `toThrowError()` so will change it now 👍